### PR TITLE
Add servant benchmark.

### PR DESCRIPTION
**Pull request by jkarni**
This commit adds benchmarks (for all test types) for the servant haskell
    web framework.

### CODE REVIEW

**Comment by jkarni** [comment]
Is it too late to have new frameworks for round 13?

**Comment by jkarni** [comment]
Build failures seem to be unrelated (`finch` and `officefloor`).

**Comment by Rydgel** [comment]
Just a warning for the plaintext test:
```
WARN for http://127.0.0.1:7041/plaintext
     Content encoding found in "text/plain;charset=utf-8" where "text/plain" is acceptable.
```

I'm not sure if it's really important.
Everything seems good tho.

**Comment by nbrady-techempower** [comment]
@jkarni Looks good. I will test it locally on Tuesday then merge it in!

**Comment by jkarni** [comment]
@Rydgel thanks. I noticed that, and may switch to simply `text/plain` in a later PR.
@nbrady-techempower thanks

File: frameworks/Haskell/servant/benchmark_config.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      "orm": "Raw",
+      "platform": "Wai",
+      "webserver": "Warp",
+      "os": "Linux",
+      "database_os": "Linux",
+      "display_name": "servant",
+      "notes": "",
+      "versus": ""
+    }
+  }]
```
Comment (nbrady-techempower) [inline frameworks/Haskell/servant/benchmark_config.json]: @jkarni would you mind updating this:
`"framework": "None"`
and also under framework add:
`"flavor": "GHC710"

I know it's a bit confusing, but framework is for whether or not your framework is using a framework. I have an open PR to start updating other benchmark configs and will be updating the docs soon.

All tests are good on local environment and will merge as soon as you can update these. Thanks!

**Comment by jkarni** [comment]
@Rydgel removed the `charset` (which actually had a very significant impact on throughput).

@nbrady-techempower updated `framework` to `None`, and `database` to `Postgres`. I've kept the new commit with changes as a separate commit so it's easier to review, but am happy to squash them.

**Comment by nbrady-techempower** [comment]
@jkarni nope, this is fine! thanks to you and your team for putting this together. I've read through some of your threads on the benchmarking and am excited to see the round 14 results. Good luck!

**Pull request merged**